hbase-server/src/main/java/org/apache/hadoop/hbase/io/hfile/HFile.java

```diff
@@ -319,7 +312,7 @@ public Writer create() throws IOException {
           LOG.debug("Unable to set drop behind on {}", path.getName());
         }
       }
-      return new HFileWriterImpl(conf, cacheConf, path, ostream, comparator, fileContext);
+      return new HFileWriterImpl(conf, cacheConf, path, ostream, fileContext);
```

Comment:
So here the comparator class name will be passed within the fileContext from upper layers. Good.

Reply:
Yes sir. Idea is that there is one place to get comparator to use. Before this patch, comparator was passed down most of the time but in encoder context and elsewhere in a few locations, comparator was hardcoded because not passed from above.